AX: [AX Thread Text Marker] Width of bounds returned from bounds for range grows as you move away from the origin

hoffmanjoshua opened the pull request:
#### 76584d15da2cb59da82653554d4b9c7e446923da
<pre>
AX: [AX Thread Text Marker] Width of bounds returned from bounds for range grows as you move away from the origin
<a href="https://bugs.webkit.org/show_bug.cgi?id=289301">https://bugs.webkit.org/show_bug.cgi?id=289301</a>
<a href="https://rdar.apple.com/146436778">rdar://146436778</a>

Reviewed by Tyler Wilcock.

When calculating the bounds for range API with the new AX thread text markers, we would pass in the location + the
length of the range to `nextMarkerFromOffset`. However, this assumes that `nextMarkerForOffset` will be starting
from a 0 offset, rather than the current text marker&apos;s offset.

This can result in text marker ranges that are much longer than expected, as they are adding the location part
of the range unnecessarily. Removing the `range.location` from this callsite fixes this behavior.

This PR also includes downgraded release asserts in `AXTextRun` to match 291629@main.

* Source/WebCore/accessibility/AXTextRun.cpp:
(WebCore::AXTextRuns::substring const):
(WebCore::AXTextRuns::domOffset const):
(WebCore::AXTextRuns::localRect const):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(-[WebAccessibilityObjectWrapper accessibilityAttributeValue:forParameter:]):

Canonical link: <a href="https://commits.webkit.org/291773@main">https://commits.webkit.org/291773@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aef8e4e6a61297887e29417922bd848001cc8aa0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/93904 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13488 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3221 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/98910 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44431 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/95954 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/13785 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/21920 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71654 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29014 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/96906 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10231 "Found 1 new test failure: fast/forms/ios/focus-input-via-button-no-scaling.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/84832 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51990 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9911 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2469 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/43747 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80175 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2551 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/100950 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/20956 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15278 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80672 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21208 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/80793 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80021 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19938 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24566 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1928 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/14113 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/20940 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/26118 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20627 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24087 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22368 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

twilco commented on Source/WebCore/accessibility/AXTextRun.cpp:
```diff
@@ -25,9 +25,20 @@
 
 #include "config.h"
 #include "AXTextRun.h"
+#import "Logging.h"
```
I think we typically use `#include` in `.cpp` files

twilco commented on Source/WebCore/accessibility/AXTextRun.cpp:
```diff
@@ -165,7 +182,9 @@ FloatRect AXTextRuns::localRect(unsigned start, unsigned end, float lineHeight)
                 unsigned fullRunWidth = run.text.length() * estimatedCharacterWidth;
 
-                RELEASE_ASSERT(fullRunWidth >= (widthPriorToStart + widthAfterEnd));
+                TEXT_RUN_ASSERT_AND_LOG(fullRunWidth >= (widthPriorToStart + widthAfterEnd), "localrect (2)");
```
Should this be capitalized like `localRect` like the method name?

hoffmanjoshua committed:
AX: [AX Thread Text Marker] Width of bounds returned from bounds for range grows as you move away from the origin
https://bugs.webkit.org/show_bug.cgi?id=289301
rdar://146436778

Reviewed by Tyler Wilcock.

When calculating the bounds for range API with the new AX thread text markers, we would pass in the location + the
length of the range to `nextMarkerFromOffset`. However, this assumes that `nextMarkerForOffset` will be starting
from a 0 offset, rather than the current text marker's offset.

This can result in text marker ranges that are much longer than expected, as they are adding the location part
of the range unnecessarily. Removing the `range.location` from this callsite fixes this behavior.

This PR also includes downgraded release asserts in `AXTextRun` to match 291629@main.

* Source/WebCore/accessibility/AXTextRun.cpp:
(WebCore::AXTextRuns::substring const):
(WebCore::AXTextRuns::domOffset const):
(WebCore::AXTextRuns::localRect const):
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm:
(-[WebAccessibilityObjectWrapper accessibilityAttributeValue:forParameter:]):

Canonical link: https://commits.webkit.org/291773@main
